x-pack/plugins/osquery/server/handlers/action/get_action_type.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { curry, uniq } from 'lodash';
+import { ActionTypeExecutorResult } from '@kbn/actions-plugin/common';
+import { osqueryActionTypeBase } from '../../../common/actions/osquery_type';
+import { createActionHandler } from './create_action_handler';
+import { OsqueryAppContext } from '../../lib/osquery_app_context_services';
+
+interface IAlert {
+  agent?: {
+    id: string;
+  };
+}
+
+export const getOsqueryActionType = (osqueryContext: OsqueryAppContext) => ({
+  ...osqueryActionTypeBase,
+  executor: curry(executor)({
+    osqueryContext,
+  }),
+});
+
+// @ts-expect-error update types
+async function executor(payload, execOptions): Promise<ActionTypeExecutorResult<unknown>> {
+  const { query, ecs_mapping: ecsMapping, alerts, id } = execOptions.params;
+
+  const parsedAlerts: IAlert[] = JSON.parse(alerts);
```

Comment:
Alerts do not have a schema yet this connectors assumes the do (`IAlert`) - this suggests this connector should only be available for use if Alerts follow a certain format.

Is this meant to be a connector only available on rules that follow the `Alerts-as-data` schema?